src/mvContext.cpp

```diff
@@ -232,269 +218,227 @@ InsertConstants_mvContext(std::vector<std::pair<std::string, long>>& constants)
     //-----------------------------------------------------------------------------
     // Mouse Codes
     //-----------------------------------------------------------------------------
-    constants.emplace_back("mvMouseButton_Left", 0);
-    constants.emplace_back("mvMouseButton_Right", 1);
-    constants.emplace_back("mvMouseButton_Middle", 2);
-    constants.emplace_back("mvMouseButton_X1", 3);
-    constants.emplace_back("mvMouseButton_X2", 4);
+    constants.emplace_back("mvMouseButton_Left", ImGuiMouseButton_Left);
+    constants.emplace_back("mvMouseButton_Right", ImGuiMouseButton_Right);
+    constants.emplace_back("mvMouseButton_Middle", ImGuiMouseButton_Middle);
+    constants.emplace_back("mvMouseButton_X1", ImGuiKey_MouseX1); // TODO: Probably not okay like this
```

Comment:
`mvMouseButton_X1`: Unfortunately the comment is right. Mouse button constants are used inside ImGui to directly index arrays of 5 elements, one per button. `ImGuiKey_MouseX1` has a numeric value beyond 512, and such assignment might break things.

One thing it definitely breaks is `add_mouse_drag_handler(dpg.mvMouseButton_X1, ...)`. It works in the old version but doesn't in the new one. I know, it's a bit of a stretch and it's rather unlikely that anyone uses this handler with X1/X2; what actually bothers me is that indexing beyond array boudaries.

So we probably should leave hardcoded 3 and 4 here.

Here's a test script that covers this and also some other mouse tests.

```py
#!/usr/local/bin/python3
import dearpygui.dearpygui as dpg

dpg.create_context()
dpg.create_viewport(title=f"Test - {dpg.get_dearpygui_version()}", width=700, height=700)
 
with dpg.handler_registry():
    dpg.add_mouse_click_handler(callback=lambda s, a: dpg.set_value("info", a))
    dpg.add_mouse_drag_handler(dpg.mvMouseButton_X1, callback=lambda s, a: dpg.set_value("drag", a))

def update_click_info():
    for btn_name in ("mvMouseButton_Left", "mvMouseButton_Right", "mvMouseButton_Middle", "mvMouseButton_X1", "mvMouseButton_X2"):
        btn_value = getattr(dpg, btn_name)
        if dpg.is_mouse_button_clicked(btn_value):
            dpg.set_value("extra", f"Clicked test returned true for: {btn_name}")

with dpg.item_handler_registry() as handlers:
    dpg.add_item_visible_handler(callback=update_click_info)
 
with dpg.window():
    dpg.bind_item_handler_registry(dpg.last_item(), handlers) 
    dpg.set_primary_window(dpg.last_item(), True)
    dpg.add_text("Click anywhere with all mouse buttons you have.\nAlso try to drag with the X1 button (the 4th button).")
    with dpg.group(horizontal=True):
        dpg.add_text("Last button clicked: ")
        dpg.add_text(tag="info")
    with dpg.group(horizontal=True):
        dpg.add_text("Drag data for X1: ")
        dpg.add_text(tag="drag")

    dpg.add_text(tag="extra")

dpg.setup_dearpygui()
dpg.show_viewport()
dpg.start_dearpygui()
dpg.destroy_context() 
```